services/order.py

```diff
@@ -0,0 +1,31 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from db.models import User, Order, MovieSession, Ticket
+
+
+def create_order(tickets: list[dict], username: str, date: str = None) -> None:
+    user = get_user_model().objects.get_or_create(username=username)
+
+    with transaction.atomic():

```

Comment:
You could place user inside of an atomic transaction and apply it as a decorator afterwards